server/entry/model.ts

```diff
@@ -0,0 +1,41 @@
+import type {Types} from 'mongoose';
+import {Schema, model} from 'mongoose';
+import type { Account } from '../account/model';
+
+type ParameterizedEntry<U, T, D> = {
+  _id: Types.ObjectId;
+  owner: U;
+  type: T;
+  detail: D;
+  notes: string;
+  date: Date;
+};
+
+export type Entry = ParameterizedEntry<Types.ObjectId, string, string>;
+export type PopulatedEntry = ParameterizedEntry<Account, string, string>;
```

Comment:
Would it be easier if I just created some sort of enum type `Type`? Also, map each `Type` to `Medication`, `Contact` etc.?

Reply:
Revisit in final